pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -23,431 +21,159 @@ const (
 	buildChainLong = `
 Output the inputs and dependencies of your builds
 
-Supported output formats are json, dot, and ast. The default is set to json.
+Supported formats for the generated graph are dot and a human-readable output.
 Tag and namespace are optional and if they are not specified, 'latest' and the 
 default namespace will be used respectively.`
 
-	buildChainExample = `  // Build dependency tree for the specified image stream and tag
-  $ openshift ex build-chain [image-stream]:[tag]
+	buildChainExample = `  // Build the dependency tree for the 'latest' tag in centos7
+  $ %[1]s build-chain centos7
 
-  // Build dependency trees for all tags in the specified image stream
-  $ openshift ex build-chain [image-stream] --all-tags
+  // Build the dependency tree for 'v2' tag in dot format and visualize it via the dot utility
+  $ %[1]s build-chain centos7:v2 -o dot | dot -T svg -o deps.svg
 
-  // Build the dependency tree using tag 'latest' in 'testing' namespace
-  $ openshift ex build-chain [image-stream] -n testing
-
-  // Build the dependency tree and output it in DOT syntax
-  $ openshift ex build-chain [image-stream] -o dot
-
-  // Build dependency trees for all image streams in the current namespace
-  $ openshift ex build-chain
-
-  // Build dependency trees for all image streams across all namespaces
-  $ openshift ex build-chain --all`
+  // Build the dependency tree across all namespaces for the specified image stream tag found in 'test' namespace
+  $ %[1]s build-chain centos7 -n test --all`
 )
 
-// Node is a representation of an image stream inside a tree
-type Node struct {
-	FullName string   `json:"fullname"`
-	Tags     []string `json:"tags,omitempty"`
-	Edges    []*Edge  `json:"edges,omitempty"`
-	Children []*Node  `json:"children,omitempty"`
-}
-
-// String helps in dumping a tree in AST format
-func (root *Node) String() string {
-	tree := ""
-	tree += root.FullName
+// BuildChainOptions contains all the options needed for build-chain
+type BuildChainOptions struct {
+	name       string
+	defaultNs  string
+	namespaces []string
+	tag        string
+	output     string
 
-	for _, n := range root.Children {
-		tree += "(" + n.String() + ")"
-	}
+	allNamespaces bool
 
-	return tree
-}
-
-// Edge represents a build configuration relationship
-// between two nodes
-//
-// Note that this type has no relation with the dot.Edge
-// type
-type Edge struct {
-	FullName string `json:"fullname"`
-	To       string `json:"to"`
-}
-
-// NewEdge adds a new edge on a parent node
-//
-// Note that this function has no relation
-// with the dot.Edge type
-func NewEdge(fullname, to string) *Edge {
-	return &Edge{
-		FullName: fullname,
-		To:       to,
-	}
+	oc client.Interface
```

Comment:
I use the client Interface for getting the specified ist in case of a "not found" error too